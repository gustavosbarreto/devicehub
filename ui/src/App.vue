<template>
<v-app>
    <v-navigation-drawer v-if="isLoggedIn" :clipped="false" fixed v-model="drawer" :mini-variant="false" enable-resize-watcher app>
        <v-container>
            <div class="text-center">
                <v-icon>mdi-console</v-icon>
                <h2 style="font-family: monospace">ShellHub</h2>
                <span class="overline">beta</span>
            </div>
        </v-container>
        <v-list>
            <v-list-item v-for="item in items" :key="item.title" :to="item.path" two-line>
                <v-list-item-action>
                    <v-icon v-text="item.icon"></v-icon>
                </v-list-item-action>

                <v-list-item-content>
                    <v-list-item-title v-text="item.title"></v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
    <v-app-bar app elevate-on-scroll color="primary" class="pl-3 pr-4" v-if="isLoggedIn">
        <v-menu transition="scale-transition" origin="top left">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-server</v-icon>
                    My Device Fleet
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>
        </v-menu>
        <v-spacer></v-spacer>
        <v-chip>
            <v-icon>help</v-icon>
        </v-chip>
        <v-chip>
            <v-icon>notifications</v-icon>
        </v-chip>
        <v-menu transition="scale-transition" origin="top right">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-account</v-icon>
                    {{ $store.getters["auth/currentUser"] }}
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>

            <v-card>
                <v-list-item three-line>
                    <v-list-item-content>
                        <v-list-item-title class="mb-1">Tenant ID</v-list-item-title>
                        <v-list-item-subtitle>
                            <v-chip>
                                <span>
                                    {{ tenant }}
                                </span>
                                <v-icon right v-clipboard="tenant" v-clipboard:success="() => { copySnack = true; }">mdi-content-copy</v-icon>
                            </v-chip>
                        </v-list-item-subtitle>
                    </v-list-item-content>
                </v-list-item>

                <v-card-actions>
                    <v-btn small text @click="logout()">
                        Logout
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-menu>
    </v-app-bar>
    <v-content>
        <v-container class="pa-8" fluid>
            <router-view></router-view>
        </v-container>
        <v-snackbar v-model="copySnack" :timeout=3000>Tenant ID copied to clipboard</v-snackbar>
    </v-content>
</v-app>
</template>

<script>
export default {
  name: "App",

  methods: {
    logout() {
      this.$store.dispatch("auth/logout").then(() => {
        this.$router.push("/login");
      });
    }
  },

  computed: {
    tenant() {
      return this.$store.getters["auth/tenant"];
    },

    isLoggedIn() {
      return this.$store.getters["auth/isLoggedIn"];
    }
  },

  data() {
    return {
      drawer: true,
      clipped: false,
      copySnack: false,
      items: [
        {
          icon: "dashboard",
          title: "Dashboard",
          path: "/"
        },
        {
          icon: "devices",
          title: "Devices",
          path: "/devices"
        },
        {
          icon: "history",
          title: "Sessions",
          path: "/sessions"
        }
      ]
    };
  }
};
</script>

<style>
.v-list-active {
  border-left: 4px solid var(--v-primary-base);
}

.text-shadow {
  text-shadow: #000 0 0 6px;
  color: transparent;
}
</style>
