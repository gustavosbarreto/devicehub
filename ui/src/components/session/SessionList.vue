<template>
<fragment>

  <div class="d-flex pa-0 align-center">
    <h1>Sessions</h1>
    <v-spacer/>
    <v-spacer/>
  </div>

  <v-card class="mt-2">
    <v-app-bar flat color="transparent">
      <v-toolbar-title></v-toolbar-title>
    </v-app-bar>
    
    <v-divider></v-divider>

    <!-- v-icon notranslate v-icon--link mdi mdi-content-copy theme--light -->
    
    <v-card-text class="pa-0">
      <v-data-table :headers="headers" :items="$store.getters['sessions/list']" item-key="uid" :sort-by="['started_at']" :sort-desc="[true]" disable-pagination hide-default-footer>
              
        <template v-slot:item.active="{ item }">
          <v-icon color="success" v-if="item.active">check_circle</v-icon>
          <v-tooltip bottom v-else>
            <template #activator="{ on }">
              <v-icon v-on="on">check_circle</v-icon>
            </template>
            <span>active {{ item.last_seen | moment("from", "now") }}</span>
          </v-tooltip>
        </template>

        <template v-slot:item.device="{ item }">
          <v-chip class="short">
            <span>{{ item.device }}</span>
            <v-icon small @click.stop v-clipboard="item.uid" v-clipboard:success="showCopySnack">mdi-content-copy</v-icon>
          </v-chip>
        </template>

        <template v-slot:item.ip_address="{ item }">
          <code>{{ item.ip_address }}</code>
        </template>

        <template v-slot:item.started="{ item }">
          {{ item.started_at | moment("ddd, MMM Do YY, h:mm:ss a")}}
        </template>

        <template v-slot:item.last_seen="{ item }">
          {{ item.last_seen | moment("ddd, MMM Do YY, h:mm:ss a")}}
        </template>

        <template v-slot:item.actions="{ item }">
          <v-icon class="icons" @click="detailsSession(item)">
            info
          </v-icon>
        </template>

      </v-data-table>
    </v-card-text>
  </v-card>
  <v-snackbar v-model="copySnack" :timeout=3000>Device UID copied to clipboard</v-snackbar>

</fragment>
</template>

<script>

export default {
  name: 'SessionList',

  data() {
    return {
      copySnack: false,

      headers: [
        {
          text: "Active",
          value: "active",
          align: "center"
        },
        {
          text: "Device",
          value: "device",
          align: "center"
        },
        {
          text: "Username",
          value: "username",
          align: "center"
        },
        {
          text: "IP Address",
          value: "ip_address",
          align: "center"
        },
        {
          text: "Started",
          value: "started",
          align: "center"
        },
        {
          text: "Last Seen",
          value: "last_seen",
          align: "center"
        },
        {
          text: "Actions",
          value: "actions",
          align: "center"
        },
      ]
    };
  },

  created() {
    this.$store.dispatch("sessions/fetch");
  },

  methods: {
    showCopySnack() {
      this.copySnack = true;
    },
    detailsSession(session){
      this.$router.push('/session/'+session.uid) 
    },
  },

};
</script>

<style>

.short{
  width:300px;
}
.short span{
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

</style>
