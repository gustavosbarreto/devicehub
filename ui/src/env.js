export default {
  /* eslint-disable no-param-reassign */
  install(Vue) {
    const env = {
      isHosted: window.env.HOSTED === 'true',
    };

    Vue.env = env;
    Vue.prototype.$env = env;
  },
  /* eslint-enable no-param-reassign */
};
