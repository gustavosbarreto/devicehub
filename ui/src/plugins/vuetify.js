import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import '@mdi/font/css/materialdesignicons.css';
import 'font-logos/assets/font-logos.css';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import { fas } from '@fortawesome/free-solid-svg-icons';
import Clipboard from 'v-clipboard';
import Fragment from 'vue-fragment';

//import 'vuetify/dist/vuetify.min.css'
//import '../styles/variables.scss'

Vue.component('font-awesome-icon', FontAwesomeIcon); // Register component globally
library.add(fas); // Include needed icons.

Vue.use(Vuetify);
Vue.use(Clipboard);
Vue.use(Fragment.Plugin);

const DEVICE_ICONS = {
  alpine: 'fl-alpine',
  arch: 'fl-archlinux',
  centos: 'fl-centos',
  coreos: 'fl-coreos',
  debian: 'fl-debian',
  devuan: 'fl-devuan',
  elementary: 'fl-elementary',
  fedora: 'fl-fedora',
  freebsd: 'fl-freebsd',
  gentoo: 'fl-gentoo',
  linuxmint: 'fl-linuxmint',
  mageia: 'fl-mageia',
  manjaro: 'fl-manjaro',
  mandriva: 'fl-mandriva',
  nixos: 'fl-nixos',
  opensuse: 'fl-opensuse',
  rhel: 'fl-redhat',
  sabayon: 'fl-sabayon',
  slackware: 'fl-slackware',
  ubuntu: 'fl-ubuntu',
  raspbian: 'fl-raspberry-pi',
  'ubuntu-core': 'fl-ubuntu',
  void: 'fl-void',
};

export default new Vuetify({
  iconfont: 'md',
  icons:{
    values:DEVICE_ICONS
  },
  theme: {
    dark: false,
    themes: {
      light: {
        primary: '#364576',
        secondary: '#b0bec5',
        accent: '#8c9eff',
        error: '#b71c1c',
      },
    }
  }
});
