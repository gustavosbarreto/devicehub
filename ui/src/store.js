import Vue from 'vue';
import Vuex from 'vuex';
import stats from '@/modules/stats';
import sessions from '@/modules/sessions';
import auth from '@/modules/auth';
import devices from '@/modules/devices';
import modals from '@/modules/modals';
import firewallrules from '@/modules/firewall_rules';
import notifications from '@/modules/notifications';

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    devices, modals, stats, sessions, auth, firewallrules, notifications,
  },
});
