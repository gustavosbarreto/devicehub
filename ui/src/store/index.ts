import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";

import { auth, AuthState } from "./modules/auth";
import { apiKeys, ApiKeysState } from "./modules/api_keys";
import { layout, LayoutState } from "./modules/layout";
import { users, UsersState } from "./modules/users";
import { tags, TagsState } from "./modules/tags";
import { stats, StatsState } from "./modules/stats";
import { spinner, SpinnerState } from "./modules/spinner";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { sessions, SessionsState } from "./modules/sessions";
import { security, SecurityState } from "./modules/security";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { privateKey, PrivateKeyState } from "./modules/private_key";
import { notifications, NotificationsState } from "./modules/notifications";
import { modal, ModalState } from "./modules/modal";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { devices, DevicesState } from "./modules/devices";
import { box, BoxState } from "./modules/box";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { billing } from "./modules/billing";
import { customer, CustomerState } from "./modules/customer";
import { announcement, AnnouncementState } from "./modules/announcement";
import apiPlugin from "./plugins/api";

export interface State {
  auth: AuthState;
  apiKeys: ApiKeysState;
  billing: NamespacesState;
  customer: CustomerState;
  box: BoxState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  modal: ModalState;
  namespaces: NamespacesState;
  notifications: NotificationsState;
  privateKey: PrivateKeyState;
  publicKeys: PublicKeysState;
  security: SecurityState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  tags: TagsState;
  users: UsersState;
  announcement: AnnouncementState;
}

export const key: InjectionKey<Store<State>> = Symbol("store");

export const store = createStore<State>({
  modules: {
    auth,
    apiKeys,
    customer,
    billing,
    box,
    devices,
    firewallRules,
    layout,
    modal,
    namespaces,
    notifications,
    privateKey,
    publicKeys,
    security,
    sessions,
    snackbar,
    spinner,
    stats,
    tags,
    users,
    announcement,
  },
  plugins: [
    apiPlugin,
  ],
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
