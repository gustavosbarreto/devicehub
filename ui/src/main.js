import Vue from 'vue';

import Snackbar from '@/components/snackbar/Snackbar';

import App from './App';
import router from './router';
import store from './store';
import env from './env';
import './vee-validate';
import vuetify from './plugins/vuetify';

Vue.config.productionTip = false;

Vue.component('Snackbar', Snackbar);

Vue.use(require('vue-moment'));

Vue.use(env);

new Vue({
  vuetify,
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
