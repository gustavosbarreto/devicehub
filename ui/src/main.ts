import { createApp } from "vue";
import * as Sentry from "@sentry/vue";
import { BrowserTracing } from "@sentry/tracing";
import { envVariables } from "./envVariables";
import vuetify from "./plugins/vuetify";
import { key, store } from "./store";
import { router } from "./router";
import App from "./App.vue";

import { loadFonts } from "./plugins/webfontloader";

import SnackbarComponent from "./components/Snackbar/Snackbar.vue";
import { SnackbarPlugin } from "./plugins/snackbar";

const app = createApp(App);

Sentry.init({
  app,
  dsn: envVariables.sentryDsn || "",
  integrations: [
    new BrowserTracing({
      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
    }),
  ],
  tracesSampleRate: 1.0,
  hooks: ["activate", "create", "destroy", "mount", "update"],
  timeout: 500,
  release: envVariables.version || "latest",
});
Sentry.setTag("project", "shellhub-ui");

loadFonts();

app.use(vuetify);
app.use(router);
app.use(store, key);
app.use(SnackbarPlugin);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
