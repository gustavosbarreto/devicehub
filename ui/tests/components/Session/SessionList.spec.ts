import { createVuetify } from "vuetify";
import { flushPromises, mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import SessionList from "../../../src/components/Sessions/SessionList.vue";
import { createStore } from "vuex";
import { key } from "../../../src/store";
import routes from "../../../src/router";

const numberSessionsGlobal = 2;
const sessionPlayShow = [false, false];
const sessionCloseShow = [false, false];

const pagination = {
  groupBy: [],
  groupDesc: [],
  itemsPerPage: 10,
  multiSort: false,
  mustSort: false,
  page: 1,
  sortBy: ["started_at"],
  sortDesc: [true],
};

const sessionsGlobal = [
  {
    uid: "8c354a00",
    device_uid: "a582b47a",
    device: {
      uid: "a582b47a",
      name: "39-5e-2a",
      identity: {
        mac: "00:00:00",
      },
      info: {
        id: "debian",
        pretty_name: "Debian",
        version: "v0.2.5",
      },
      public_key: "xxxxxxxx",
      tenant_id: "00000000",
      last_seen: "2020-05-18T13:27:02.498Z",
      online: false,
      namespace: "user",
    },
    tenant_id: "00000000",
    username: "user",
    ip_address: "00.00.00",
    started_at: "2020-05-18T12:30:28.824Z",
    last_seen: "2020-05-18T12:30:30.205Z",
    active: true,
    authenticated: false,
  },
  {
    uid: "8c354a01",
    device_uid: "a582b47a",
    device: {
      uid: "a582b47a",
      name: "b4-2e-99",
      identity: {
        mac: "00:00:00",
      },
      info: {
        id: "debian",
        pretty_name: "Debian",
        version: "v0.2.5",
      },
      public_key: "xxxxxxxx",
      tenant_id: "00000000",
      last_seen: "2020-05-18T13:27:02.498Z",
      online: false,
      namespace: "user",
    },
    tenant_id: "00000000",
    username: "user",
    ip_address: "00.00.00",
    started_at: "2020-05-18T12:30:28.824Z",
    last_seen: "2020-05-18T12:30:30.205Z",
    active: false,
    authenticated: false,
  },
];

const headers = [
  {
    text: "Active",
    value: "active",
  },
  {
    text: "Device",
    value: "device",
  },
  {
    text: "Username",
    value: "username",
  },
  {
    text: "Authenticated",
    value: "authenticated",
  },
  {
    text: "IP Address",
    value: "ip_address",
  },
  {
    text: "Started",
    value: "started",
  },
  {
    text: "Last Seen",
    value: "last_seen",
  },
  {
    text: "Actions",
    value: "actions",
  },
];

const tests = [
  {
    description: "Sessions has enterprise version when user has owner role",
    role: {
      type: "owner",
      permission: true,
    },
    variables: {
      sessions: sessionsGlobal,
      numberSessions: numberSessionsGlobal,
      enterprise: true,
    },
    data: {
      menu: false,
      loading: false,
      itemsPerPage: 10,
      page: 1,
      pagination,
      headers,
    },
    computed: {
      getListSessions: sessionsGlobal,
      getNumberSessions: numberSessionsGlobal,
      isEnterprise: true,
      hasAuthorizationPlay: true,
    },
  },
  {
    description: "Sessions has enterprise version when user has observer role",
    role: {
      type: "observer",
      permission: false,
    },
    variables: {
      sessions: sessionsGlobal,
      numberSessions: numberSessionsGlobal,
      enterprise: true,
    },
    data: {
      menu: false,
      loading: false,
      itemsPerPage: 10,
      page: 1,
      pagination,
      headers,
    },
    computed: {
      getListSessions: sessionsGlobal,
      getNumberSessions: numberSessionsGlobal,
      isEnterprise: true,
      hasAuthorizationPlay: false,
    },
  },
  {
    description: "Sessions has no enterprise version",
    role: {
      type: "owner",
      permission: true,
    },
    variables: {
      sessions: sessionsGlobal,
      numberSessions: numberSessionsGlobal,
      enterprise: false,
    },
    data: {
      pagination,
      menu: false,
      loading: false,
      itemsPerPage: 10,
      page: 1,
      headers,
    },
    computed: {
      getListSessions: sessionsGlobal,
      getNumberSessions: numberSessionsGlobal,
      isEnterprise: false,
      hasAuthorizationPlay: true,
    },
  },
];

const store = (sessions: any, numberSessions: any, currentRole: any) => {
  return createStore({
    state: {
      sessions,
      numberSessions,
      currentRole,
    },
    getters: {
      "sessions/list": (state) => state.sessions,
      "sessions/getNumberSessions": (state) => state.numberSessions,
      "box/getStatus": () => true,
      "auth/role": (state) => state.currentRole,
    },
    actions: {
      "sessions/fetch": vi.fn(),
      "sessions/close": vi.fn(),
      "snackbar/showSnackbarErrorAssociation": vi.fn(),
      "snackbar/showSnackbarErrorLoading": vi.fn(),
      "snackbar/setSnackbarErrorDefault": vi.fn(),
      "box/setStatus": vi.fn(),
    },
  });
};
describe("SessionList", () => {
  let wrapper: VueWrapper<any>;
  const vuetify = createVuetify();

  tests.forEach((test) => {
    describe(`${test.description}`, () => {
      beforeEach(() => {
        wrapper = mount(SessionList, {
          global: {
            plugins: [
              [
                store(
                  test.variables.sessions,
                  test.variables.numberSessions,
                  test.role.type
                ),
                key,
              ],
              routes,
              vuetify,
            ],
          },
          shallow: true,
        });
      });

      ///////
      // Component Rendering
      //////

      it("Is a Vue instance", () => {
        expect(wrapper).toBeTruthy();
      });
      it("Renders the component", () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

      ///////
      // Data checking
      //////
      it("Data is defined", () => {
        expect(wrapper.vm.$data).toBeDefined();
      });
      it("Receive props in props", () => {
        expect(wrapper.vm.loading).toBe(test.data.loading);
        expect(wrapper.vm.itemsPerPage).toBe(test.data.itemsPerPage);
        expect(wrapper.vm.page).toBe(test.data.page);
        expect(wrapper.vm.headers).toEqual(test.data.headers);
      });
      it("Check the computed", () => {
        expect(wrapper.vm.sessions).toEqual(test.computed.getListSessions);
        expect(wrapper.vm.numberSessions).toEqual(
          test.computed.getNumberSessions
        );
      });

      //////
      // HTML validation
      //////

      it("Renders the template with data", () => {
        expect(
          wrapper.find('[data-test="sessions-list"]').exists()
        ).toBeTruthy();
      });
    });
  });
});
