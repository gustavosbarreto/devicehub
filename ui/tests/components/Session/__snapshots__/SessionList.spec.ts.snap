// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SessionList > Sessions has enterprise version when user has observer role > Renders the component 1`] = `
"<div data-v-459ab1b4=\\"\\">
  <data-table-stub data-v-459ab1b4=\\"\\" itemsperpage=\\"10\\" totalcount=\\"2\\" headers=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" items=\\"[object Object],[object Object]\\" comboboxoptions=\\"10,20,50,100\\" loading=\\"false\\" actualpage=\\"1\\" nextpage=\\"[Function]\\" previouspage=\\"[Function]\\" data-test=\\"sessions-list\\"></data-table-stub>
</div>"
`;

exports[`SessionList > Sessions has enterprise version when user has owner role > Renders the component 1`] = `
"<div data-v-459ab1b4=\\"\\">
  <data-table-stub data-v-459ab1b4=\\"\\" itemsperpage=\\"10\\" totalcount=\\"2\\" headers=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" items=\\"[object Object],[object Object]\\" comboboxoptions=\\"10,20,50,100\\" loading=\\"false\\" actualpage=\\"1\\" nextpage=\\"[Function]\\" previouspage=\\"[Function]\\" data-test=\\"sessions-list\\"></data-table-stub>
</div>"
`;

exports[`SessionList > Sessions has no enterprise version > Renders the component 1`] = `
"<div data-v-459ab1b4=\\"\\">
  <data-table-stub data-v-459ab1b4=\\"\\" itemsperpage=\\"10\\" totalcount=\\"2\\" headers=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" items=\\"[object Object],[object Object]\\" comboboxoptions=\\"10,20,50,100\\" loading=\\"false\\" actualpage=\\"1\\" nextpage=\\"[Function]\\" previouspage=\\"[Function]\\" data-test=\\"sessions-list\\"></data-table-stub>
</div>"
`;
